go/signedexchange/cmd/dump-signedexchange/main.go

```diff
@@ -5,50 +5,81 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"net/http"
 	"os"
 	"time"
 
 	"github.com/WICG/webpackage/go/signedexchange"
 )
 
 var (
-	flagInput     = flag.String("i", "", "Signed-exchange input file")
-	flagSignature = flag.Bool("signature", false, "Print only signature value")
-	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
 	flagCert      = flag.String("cert", "", "Certificate CBOR file. If specified, used instead of fetching from signature's cert-url")
+	flagHeaders   = flag.Bool("headers", true, "Print headers")
+	flagFilename  = flag.String("i", "", "Signed-exchange input file")
+	flagPayload   = flag.Bool("payload", true, "Print payload")
+	flagSignature = flag.Bool("signature", false, "Prints only signature value")
+	flagURI       = flag.String("uri", "", "Signed-exchange uri")
+	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
 )
 
 func run() error {
-	in := os.Stdin
-	if *flagInput != "" {
-		var err error
-		in, err = os.Open(*flagInput)
+	fi, err := os.Stdin.Stat()
+	var e *signedexchange.Exchange
+	if *flagFilename != "" {
+		in, err := os.Open(*flagFilename)
 		if err != nil {
 			return err
 		}
 		defer in.Close()
+		e, err = signedexchange.ReadExchange(in)
+		if err != nil {
+			return err
+		}
+	} else if *flagURI != "" {
+		client := http.DefaultClient
+		req, err := http.NewRequest("GET", *flagURI, nil)
+		req.Header.Add("Accept", "application/signed-exchange;v=b3")
+		req.Header.Add("AMP-Cache-Transform", "google;v=1")
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		e, err = signedexchange.ReadExchange(resp.Body)
+		if err != nil {
+			return err
+		}
+	} else if (fi.Mode() & os.ModeCharDevice) == 0 {
```

Comment:
I think we can rely on `strings.TrimSpace(*flagFilename) == "-"` instead

Reply:
Either works for me, but I like the current approach the best. Allows pipe usage to work without specifying `-i` at all, while still presenting the usage info when not in a pipe.

Reply:
I've kept this as is, input can come from the pipe, the -i flag or the -uri flag